### PR TITLE
[30027] Fix onEnter bodyClasses to be used instead of controller classes

### PR DESCRIPTION
**Pull request by oliverguenther**
https://community.openproject.com/wp/30027

### CODE REVIEW

File: frontend/src/app/modules/calendar/wp-calendar/wp-calendar.component.ts [hunk 1]
```diff
@@ -92,6 +92,7 @@ export class WorkPackagesCalendarController implements OnInit, OnDestroy {
     jQuery(element).tooltip({
       content: this.contentString(workPackage),
       items: '.fc-content',
+      close: function () { jQuery(".ui-helper-hidden-accessible").remove(); },
```
Comment (oliverguenther) [inline frontend/src/app/modules/calendar/wp-calendar/wp-calendar.component.ts:95]: Unrelated fix. tooltips keep spamming aria-live regions in the DOM

Comment (HDinger) [reply frontend/src/app/modules/calendar/wp-calendar/wp-calendar.component.ts:95]: 👍

**Review by HDinger — COMMENTED** [review]
@oliverguenther looks good overall. There is however still a different scroll behaviour depending on where you come from: 
When coming from the calendar, you scroll the complete page. When coming for example from the table you can scroll the WP content and the activity pane independently. However this is rather low priority as the scrolling itself works

**Comment by HDinger** [comment]
Fixes also https://community.openproject.com/projects/openproject/work_packages/29914/activity

**Comment by oliverguenther** [comment]
@HDinger this was caused by the `data` attribute being inherited by parent states. That's why the calendar state had the `router--work-packages-base` class defined, which got removed upon entering the full view state. This can be fixed by ensuring each substate of work-packages has a bodyClass defined (it could be 'undefined' if none should be applied). https://github.com/opf/openproject/pull/7225/files#diff-33e4e14f68e45d4a5ecd27ea967932f8R47
